File: service/src/main/java/com/service/domain/KrBankData.java
package com.service.domain;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

@Entity
@NoArgsConstructor
@Table(name = "krbankdata")
@Document
@Data
public class KrBankData    {
        @Column(name = "id")
        @Id
        @GeneratedValue
        Long id;

        @SerializedName("UNIT_NAME")  @Expose @Column(name = "UNIT_NAME")         String unitName;
        @SerializedName("STAT_NAME")  @Expose @Column(name = "STAT_NAME")         String statName;
        @SerializedName("STAT_CODE")  @Expose @Column(name = "STAT_CODE")         String statCode;
        @SerializedName("ITEM_CODE1") @Expose @Column(name = "ITEM_CODE1")         String itemCode1;
        @SerializedName("ITEM_CODE2") @Expose @Column(name = "ITEM_CODE2")         String itemCode2;
        @SerializedName("ITEM_CODE3") @Expose @Column(name = "ITEM_CODE3")         String itemCode3;
        @SerializedName("ITEM_NAME1") @Expose @Column(name = "ITEM_NAME1")         String itemName1;
        @SerializedName("ITEM_NAME2") @Expose @Column(name = "ITEM_NAME2")         String itemName2;
        @SerializedName("ITEM_NAME3") @Expose @Column(name = "ITEM_NAME3")         String itemName3;
        @SerializedName("DATA_VALUE") @Expose @Column(name = "DATA_VALUE")         String dataValue;
        @SerializedName("TIME")       @Expose @Column(name = "createdDate")        String  createdDate;
}

File: src/main/kotlin/com/minssan9/financial/krbank/domain/KrBankData.kt
package com.minssan9.financial.krbank.domain

import com.google.gson.annotations.SerializedName
import lombok.NoArgsConstructor
import javax.persistence.*

@Entity
@NoArgsConstructor
@Table(name = "krbankdata")
data class KrBankData(
        @Column(name = "id")
        @Id @GeneratedValue(strategy = GenerationType.AUTO)
        val id : Int,

        @Column(name = "UNIT_NAME") @SerializedName("UNIT_NAME") val unit_name : String,
        @Column(name = "STAT_NAME") @SerializedName("STAT_NAME") val stat_name : String,
        @Column(name = "ITEM_CODE1") @SerializedName("ITEM_CODE1") val item_code1 : String,
        @Column(name = "STAT_CODE") @SerializedName("STAT_CODE") val stat_code : String,
        @Column(name = "ITEM_CODE2") @SerializedName("ITEM_CODE2") val item_code2 : String,
        @Column(name = "ITEM_CODE3") @SerializedName("ITEM_CODE3") val item_code3 : String,
        @Column(name = "ITEM_NAME1") @SerializedName("ITEM_NAME1") val item_name1 : String,
        @Column(name = "ITEM_NAME2") @SerializedName("ITEM_NAME2") val item_name2 : String,
        @Column(name = "DATA_VALUE") @SerializedName("DATA_VALUE") val data_value : Double,
        @Column(name = "ITEM_NAME3") @SerializedName("ITEM_NAME3") val item_name3 : String,
        @Column(name = "TIME") @SerializedName("TIME") val time : String

)

//@Entity
//data class StatisticSearchRow(
//        @Id @GeneratedValue
//        val id : Long,
//
//        @SerializedName("UNIT_NAME") val unit_name : String,
//        @SerializedName("STAT_NAME") val stat_name : String,
//        @SerializedName("ITEM_CODE1") val item_code1 : String,
//        @SerializedName("STAT_CODE") val stat_code : String,
//        @SerializedName("ITEM_CODE2") val item_code2 : String,
//        @SerializedName("ITEM_CODE3") val item_code3 : String,
//        @SerializedName("ITEM_NAME1") val item_name1 : String,
//        @SerializedName("ITEM_NAME2") val item_name2 : String,
//        @SerializedName("DATA_VALUE") val data_value : Double,
//        @SerializedName("ITEM_NAME3") val item_name3 : String,
//        @SerializedName("TIME") val time : String
//)
